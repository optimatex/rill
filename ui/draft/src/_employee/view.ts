export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
