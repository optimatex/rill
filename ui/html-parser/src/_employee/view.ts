export { React, RC } from '@rill/view';
