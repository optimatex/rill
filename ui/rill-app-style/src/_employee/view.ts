export { createGlobalStyle, css } from '@rill/view';
