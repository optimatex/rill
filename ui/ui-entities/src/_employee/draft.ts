export { TextEditor } from '@rill/draft';
