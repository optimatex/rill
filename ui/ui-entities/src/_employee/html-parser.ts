export { htmlParser } from '@rill/html-parser';
