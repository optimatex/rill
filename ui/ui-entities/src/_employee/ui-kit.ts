export {
  // Layout
  Row,
  ListContainer,
  // Card
  Card,
  Post,
  Panel,
  // Menu
  MenuItemLink,
  // Typography
  Paragraph,
  Name,
  // Image
  Image,
  ImageCropper,
  ImagesLayout,
  // Video
  YoutubeVideos,
  // Content
  RichContentContainer,
} from '@rill/ui-kit';
