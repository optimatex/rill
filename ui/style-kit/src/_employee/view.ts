import { css as originalCss } from '@rill/view';

export const css = originalCss;
