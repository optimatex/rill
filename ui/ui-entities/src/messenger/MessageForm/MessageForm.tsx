import { React } from '_employee/view';
import { TextEditor } from '_employee/draft';
import { Panel } from '_employee/ui-kit';

import { OnAddMessage } from '@types';

interface IProps {
  onAddMessage: OnAddMessage;
}

const MessageForm = ({ onAddMessage }: IProps) => (
  <Panel border="top">
    <TextEditor
      clearOnSubmit
      onSubmit={onAddMessage}
      placeholder="Type your message here..."
    />
  </Panel>
);

export default MessageForm;
