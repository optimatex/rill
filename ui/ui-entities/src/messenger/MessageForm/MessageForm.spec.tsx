import { React } from '_employee/view';
import { render } from 'test';
import MessageForm from '.';

describe('MessageForm', () => {
  it('should render without errors', () => {
    const { container } = render(<MessageForm onAddMessage={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
