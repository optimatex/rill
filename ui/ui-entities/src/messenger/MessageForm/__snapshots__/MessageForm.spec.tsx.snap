// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should render without errors 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  padding: 4px 8px;
  border-width: 1px;
  border-style: solid;
  background: red;
  color: white;
  padding-left: 8px;
  padding-right: 8px;
}

.c0 {
  border-width: 1px;
  border-style: solid;
  border-width: 0;
  border-top-width: 1px;
}

.c1 {
  width: 100%;
  padding: 10px;
}

.c1 .public-DraftEditor-content {
  margin-bottom: 10px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="text-editor-wrap"
    >
      <div
        class="DraftEditor-root"
      >
        <div
          class="public-DraftEditorPlaceholder-root"
        >
          <div
            class="public-DraftEditorPlaceholder-inner"
            id="placeholder-1"
            style="white-space: pre-wrap;"
          >
            Type your message here...
          </div>
        </div>
        <div
          class="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-1"
            class="notranslate public-DraftEditor-content"
            contenteditable="true"
            role="textbox"
            spellcheck="false"
            style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
          >
            <div
              data-contents="true"
            >
              <div
                class=""
                data-block="true"
                data-editor="1"
                data-offset-key="1-0-0"
              >
                <div
                  class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="1-0-0"
                >
                  <span
                    data-offset-key="1-0-0"
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="c2"
      >
        Send
      </button>
    </div>
  </div>
</div>
`;
